language/move-core/types/src/state.rs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::cell::RefCell;
+
+#[derive(Clone, Copy, Debug)]
+pub enum VMState {
+    DESERIALIZER,
+    VERIFIER,
+    RUNTIME,
```

Comment:
this is currently unused and I assume you have it here in case we may want to have runtime catches?
Wonder fi we want it at all, question for you (@chen-robert) and everybody else

Reply:
Yeah, added it in for runtime catches, might make sense to remove for now and add it back later.